consensus/src/pipeline/virtual_processor/processor.rs

```diff
@@ -303,11 +311,19 @@ impl VirtualStateProcessor {
             .expect("all possible rule errors are unexpected here");
 
         // Update the pruning processor about the virtual state change
-        let sink_ghostdag_data = self.ghostdag_store.get_compact_data(new_sink).unwrap();
+        let compact_sink_ghostdag_data = if prev_sink != new_sink {
+            // we need to check with full data here, since we may need to update the window caches
+            let sink_ghostdag_data = self.ghostdag_store.get_data(new_sink).unwrap();
+            // update window caches - for ibd performance. see method comment for more details.
+            self.maybe_commit_windows(new_sink, &sink_ghostdag_data);
```

Comment:
1. It seems like the call to `maybe_commit_windows` should take place before the call `calculate_and_commit_virtual_state` where virtual DAA and median time windows are built (so we'll get 100% cache hits for virtual's selected parent)
2. Let's simply call it `cache_windows`. Then it will be obvious that if these windows are already cached it's a no op.